phantoms/MR_XCAT_qMRI/sim_ivim_sig.py

```diff
@@ -366,76 +368,123 @@ def XCAT_to_MR_DCE(XCAT, TR, TE, bvalue, D, f, Ds, b0=3, ivim_cont = True):
     return MR, Dim, fim, Dpim, legend
 
 if __name__ == '__main__':
-    bvalue = np.array([0., 1, 2, 5, 10, 20, 30, 50, 75, 100, 150, 250, 350, 400, 550, 700, 850, 1000])
-    noise = 0.0005
-    motion = False
-    interleaved = False
-    sig, XCAT, Dim, fim, Dpim, legend = phantom(bvalue, noise, motion=motion, interleaved=interleaved)
-    # sig = np.flip(sig,axis=0)
-    # sig = np.flip(sig,axis=1)
-    res=np.eye(4)
-    res[2]=2
+    parser = argparse.ArgumentParser(description=f"""
+    A commandline for generating a 4D IVIM phantom as nifti file
+                                    """)
 
-    voxel_selector_fraction = 0.5
-    D, f, Ds = contrast_curve_calc()
-    ignore = np.isnan(D)
-    generic_data = {}
-    for level, name in legend.items():
-        if len(ignore) > level and ignore[level]:
-            continue
-        selector = XCAT == level
-        voxels = sig[selector]
-        if len(voxels) < 1:
-            continue
-        signals = np.squeeze(voxels[int(voxels.shape[0] * voxel_selector_fraction)]).tolist()
-        generic_data[name] = {
-            'noise': noise,
-            'D': np.mean(Dim[selector], axis=0),
-            'f': np.mean(fim[selector], axis=0),
-            'Dp': np.mean(Dpim[selector], axis=0),
-            'data': signals
-        }
-    generic_data['config'] = {
-        'bvalues': bvalue.tolist()
-    }
-    with open('generic.json', 'w') as f:
-        json.dump(generic_data, f, indent=4)
+    def parse_bvalues_file(file_path):
+        """Used for passing the JSON file"""
+        if not os.path.exists(file_path):
+            raise argparse.ArgumentTypeError(f"File '{file_path}' does not exist")
+
+        try:
+            with open(file_path, "r") as file:
+                bvalues_dict = json.load(file)
+                if not isinstance(bvalues_dict, dict):
+                    raise argparse.ArgumentTypeError("JSON file does not contain a dict of b-values")
+                for _, bvalue in bvalues_dict.items():
+                    if not isinstance(bvalue, list):
+                        raise argparse.ArgumentTypeError("bvalues in JSON file are not list")
+                    for value in bvalue:
+                        if not isinstance(value, float):
+                            raise argparse.ArgumentTypeError("Values in lists are not float")
+        except json.JSONDecodeError as e:
+            raise argparse.ArgumentTypeError(f"Invalid JSON file: {e}")
+
+        return bvalues_dict
 
+    parser.add_argument("-b", "--bvalue", type=float,
```

Comment:
Perhaps default to the file b_values.json and the "original" set. Then the if/else below can also be removed.

Reply:
I think there wil still be if/else. It will look something like this.

`
 bvalues = None
    if args.bvalues_file:
        bvalues = args.bvalues_file
    elif args.bvalue:
        bvalues = {"cmd": args.bvalue}
    else:
       bvalues = parse_bvalues_file("b_values.json")
`

If we have default for both of the value(-f, -b), they will both give error due to :- 

`
    if args.bvalues_file and args.bvalue:
        raise argparse.ArgumentError(None, "Arguments --bvalues-file and --bvalues are mutually exclusive")
`

Reply:
Ah, you're right. Maybe just default to using the "original" set from the file then? It seems the most standard.